satpy/readers/viirs_edr.py

```diff
@@ -215,6 +215,7 @@ def available_datasets(self, configured_datasets=None):
             if self.file_type_matches(ds_info["file_type"]) is None:
                 # this is not the file type for this dataset
                 yield None, ds_info
+                continue
```

Comment:
I'm probably missing something, but what does this have to do with modis l3?

Reply:
Nothing, but we noticed that bug while doing this PR - so Dave suggested I fix it.
[i.e: I'm blaming Dave ;-) ]

Reply:
Technically the VIIRS EDR reader should probably have a test for this. I could add to this PR and copy what @simonrp84 did for his test of `available_datasets` since that seems like one of the only ways to verify this (as it does generally depend on the order of file types being processed and what's in the YAML file).